runtime-light/tl/tl-types.h

```diff
@@ -262,6 +262,27 @@ struct Dictionary final {
   }
 };
 
+struct String {
+  string value;
+
+  bool fetch(TLBuffer &tlb) noexcept {
+    if (tlb.lookup_trivial<uint32_t>().value_or(TL_ZERO) != TL_STRING) {
```

Comment:
You can use `fetch_trivial` here. `fetch` is assumed to fetch something from TL buffer. It may fail, leaving the buffer in invalid state. There is distinct functionality to save/restore TL buffer's state.

Reply:
+